components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/PublicClientAuthenticator.java

```diff
@@ -80,6 +81,25 @@ public boolean authenticateClient(HttpServletRequest request, Map<String, List>
     public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
             context) {
 
+        List publicClientSupportedGrantTypes = OAuthServerConfiguration.getInstance().
+                getPublicClientSupportedGrantTypesList();
+        List grantTypes = bodyParams.get("grant_type");
+        boolean publicClientNotSupportedGrantDetected = false;
+
+        if (publicClientSupportedGrantTypes != null && grantTypes != null) {
+            for (Object grantType : grantTypes) {
+                if (!publicClientSupportedGrantTypes.contains(grantType)) {
+                    log.error("Grant type : " + grantType + " is not supported for public clients.");
```

Comment:
log.warn("The request has sent with a Grant type : " + grantType + ", which is not allowed for public clients.");

Reply:
Changed